Content.Server/Chat/Systems/ChatSystem.Emote.cs

```diff
@@ -159,17 +170,18 @@ public bool TryPlayEmoteSound(EntityUid uid, EmoteSoundsPrototype? proto, string
     /// </summary>
     /// <param name="uid"></param>
     /// <param name="textInput"></param>
-    private void TryEmoteChatInput(EntityUid uid, string textInput)
+    /// <returns>True if an emote was performed. False if no emote was found in the text, if an emote was found but unavailable, if the emote's completion was cancelled, etc.</returns>
+    private bool TryEmoteChatInput(EntityUid uid, string textInput)
     {
         var actionTrimmedLower = TrimPunctuation(textInput.ToLower());
         if (!_wordEmoteDict.TryGetValue(actionTrimmedLower, out var emote))
-            return;
+            return false;
 
         if (!AllowedToUseEmote(uid, emote))
-            return;
+            return false;
 
-        InvokeEmoteEvent(uid, emote);
-        return;
+        // Don't show a popup on failure since the emote text will go through regardless.
+        return InvokeEmoteEvent(uid, emote, showPopupOnFailure: false);
```

Comment:
This returns the incorrect value (See other comment). I would also recommend either making this show the popup but also block the chat message (Either that or make the scream button not block the chat message either).

Look in `ChatSystem.cs:596` and just don't send the message in the chat if the emote was blocked (I was able to do it, its only a couple of lines)!

Reply:
My thinking was that since you can attempt emotes you can't perform via `/me`, this should work the same way.
ie. a mothroach doing `/me laugh` still shows `The Mothroach laugh` without making any noise; so a muzzled human doing `/me laugh` should still show `Urist laugh` without making noise.

I'll swap it as you say, though.

Reply:
> 

My thinking was that since you can attempt emotes you can't perform via /me, this should work the same way.
ie. a mothroach doing /me laugh still shows The Mothroach laugh without making any noise; so a muzzled human doing /me laugh should still show Urist laugh without making noise.

Yeah if you want to make the emote still show up with text I think that's fine and its totally up to you, just make sure its consistent like I said before. It feels weird if the scream button wont show it in text chat but typing it in will.